src/selectors/entities/channels.js

```diff
@@ -442,16 +443,16 @@ export const getSortedUnreadChannelIds = createIdsSelector(
                 bIsMention = true;
             }
 
-            if (aIsMention === bIsMention) {
+            if (aIsMention === bIsMention && isChannelMuted(bMember) === isChannelMuted(aMember)) {
                 return sortChannelsByDisplayName(locale, a, b);
-            } else if (aIsMention) {
+            } else if (aIsMention || (isChannelMuted(bMember) && !isChannelMuted(aMember))) {
```

Comment:
I think this conditions are wrong. For example, if I have two channels, both with mentions, one muted and the other not, the first `a` channel will be always the first one, independently which one is muted or which name it has.

Reply:
I'm not able to reproduce the problem, so I will approve this :)